src/app/loadout-builder/generated-sets/utils.ts

```diff
@@ -165,35 +176,112 @@ function canAllModsBeUsed(set: ArmorSet, seasonalMods: readonly LockedModBase[])
   return false;
 }
 
+/**
+ * Checks that all the general mods can fit in a set, including the energy specific ones
+ * i.e. Void Resist ect
+ */
+function canAllGeneralModsBeUsed(generalMods: readonly LockedArmor2Mod[], set: ArmorSet): boolean {
+  let armour2Count = set.firstValidSet.filter((item) => item.isDestiny2() && item.energy).length;
+
+  if (generalMods && armour2Count < generalMods.length) {
+    return false;
+  }
+
+  const generalModsByEnergyType = _.keyBy(
+    _.groupBy(generalMods, (mod) => mod.mod.plug.energyCost.energyType),
+    (group) => group[0].mod.plug.energyCost.energyType
```

Comment:
It's unclear to me why you'd nest a keyBy and a groupBy. groupBy is already returning an object keyed by your function's result.

Reply:
I had in my head that groupBy just grouped into an array of arrays for some reason. I should have checked the spec.